### ISSUE TITLE
Many issues after updating WC to 4.7.0

### ISSUE THREAD

**Issue by csimpi** [TycheSoftwares/custom-order-statuses-woocommerce#52]
**Describe the bug**

I'm getting a lot of issues with the plugin after updating Woocommerce to 4.7.0.
For some reason when I change the status manually in the dashboard (Woocommerce -> Orders -> Select an Order -> Change status -> Click Save ), in the Orders list the slug appears not the name of the custom status.

![image](https://user-images.githubusercontent.com/6192067/100546354-42de1800-3261-11eb-8c02-6db0036d1325.png)

`woocommerce_order_status_changed` doesn't fire anymore at all when I change status in WC Dashboard the same way. I guess this is related to the previous issue above. 

For some reason now the `wc-` prefix is mandatory despite that nothing changed regarding this. 

I can't add new statuses, it's complaining about the icon format which is not determined what should it be. When I click to the link and copy paste the name/slug of the icon it doesn't accept it. I had to go back and copy paste an icon code from the previous version :/ 

I just printed the statuses programmatically and I'm getting `wc-wc-` prefixes. When I remove `wc-` before the slug in dashboard all of the custom statuses disappear. ://// 

So this new version is terribly buggy, please fix these ASAP, cause our complete ordering method is hanging now. 

**To Reproduce**
Woocommerce -> Orders -> Select an Order -> Change status -> Click Save

**Expected behavior**

Works the same way as before (described above)

**Screenshots**
Above

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Win 10
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] newest

**Smartphone (please complete the following information):**

**Additional context**

### UPDATE
It seems this is a confirmed bug:

When I use the free version the email section is disabled BUT the email settings is ENABLED by default some reason, and not possible to disable. 

In your code `class-alg-wc-custom-order-statuses-core`

```
			// Emails.
			if ( 'yes' === get_option( 'alg_orders_custom_statuses_emails_enabled', 'no' ) ) {
				add_action( 'woocommerce_order_status_changed', array( $this, 'send_email_on_order_status_changed' ), PHP_INT_MAX, 4 );
			}
```

![image](https://user-images.githubusercontent.com/6192067/100546609-94d36d80-3262-11eb-8cb9-b2361f6c242c.png)

So it is clear, you override the `woocommerce_order_status_changed` hook for no reason since the user uses the free version. Would be smarter to remove these sections from your code because as I can see anybody can use the free version as premium with 5 minutes of coding.

**Comment by nitin-ts** [comment]
Hello @csimpi 
sorry for the issues you are facing with our plugin,
we have checked your issues and it is working fine in our end.

Will you please let us know about your issues in breif?

**For some reason when I change the status manually in the dashboard (Woocommerce -> Orders -> Select an Order -> Change status -> Click Save ), in the Orders list the slug appears not the name of the custom status.**
=> this is working fine in our end.
https://i.imgur.com/D682ZE3.png
https://i.imgur.com/cGyN1Bm.png

**For some reason now the wc- prefix is mandatory despite that nothing changed regarding this.**
=> no, it is not mandatory to add wc- prefix to anywhere. wc- prefix is added by default from the backend

**I can't add new statuses, it's complaining about the icon format which is not determined what should it be. When I click to the link and copy paste the name/slug of the icon it doesn't accept it. I had to go back and copy paste an icon code from the previous version :/**
=> you must not use icon slug, and only use icon code ( means e000 for this icon => https://i.imgur.com/OJdcW1V.png ) 

**When I use the free version the email section is disabled BUT the email settings is ENABLED by default some reason, and not possible to disable.**
=> in `class-alg-wc-custom-order-statuses-core` file, that code is for sending email for the custom statuses. You can enable that setting from here ( https://i.imgur.com/Fql28ss.png ), so you can not use the premium email setting even if you unblock the premium version setting, in premium version, we have the functionality to set different emails on each custom status, whereas in lite version, you can set a single email for all statuses.

We would be happy to look at this issue on your site if you can give us the site admin details.

Thanks.

**Comment by ashokrane** [comment]
@csimpi Closing this issue as we haven't received a response. Please feel free to re-open it or add a new issue if you face any problems.
